src/handlers/annotate.ts

```diff
@@ -0,0 +1,157 @@
+      await handleSimilarIssuesComment(context, payload, commentBody, comment.id, processedIssues);
+      return;
+    }
+  }
+  context.logger.info("No similar issues found for comment", { commentBody });
+}
+
+function filterByScope(scope: string, repoOrg: string, similarIssueRepoOrg: string, repoName: string, similarIssueRepoName: string): boolean {
+  switch (scope) {
+    case "global":
+      return true;
+    case "org":
+      return repoOrg === similarIssueRepoOrg;
+    case "repo":
+      return repoOrg === similarIssueRepoOrg && repoName === similarIssueRepoName;
+    default:
+      return false;
+  }
+}
+
+async function handleSimilarIssuesComment(
+  context: Context,
+  payload: Context["payload"],
+  commentBody: string,
+  commentId: number,
+  issueList: IssueGraphqlResponse[]
+) {
+  // Find existing footnotes in the body
+  const footnoteRegex = /\[\^(\d+)\^\]/g;
+  const existingFootnotes = commentBody.match(footnoteRegex) || [];
+  const highestFootnoteIndex = existingFootnotes.length > 0 ? Math.max(...existingFootnotes.map((fn) => parseInt(fn.match(/\d+/)?.[0] ?? "0"))) : 0;
+  let updatedBody = commentBody;
+  let footnotes: string[] | undefined;
+  // Sort relevant issues by similarity in ascending order
+  issueList.sort((a, b) => parseFloat(a.similarity) - parseFloat(b.similarity));
+  issueList.forEach((issue, index) => {
+    const footnoteIndex = highestFootnoteIndex + index + 1; // Continue numbering from the highest existing footnote number
+    const footnoteRef = `[^0${footnoteIndex}^]`;
+    const modifiedUrl = issue.node.url.replace("https://github.com", "https://www.github.com");
+    const { sentence } = issue.mostSimilarSentence;
+    // Insert footnote reference in the body
+    const sentencePattern = new RegExp(`${sentence.replace(/[.*+?^${}()|[\]\\]/g, "\\$&")}`, "g");
+    updatedBody = updatedBody.replace(sentencePattern, `${sentence} ${footnoteRef}`);
+
+    // Initialize footnotes array if not already done
+    if (!footnotes) {
+      footnotes = [];
+    }
+    // Add new footnote to the array
+    footnotes.push(`${footnoteRef}: ${issue.similarity}% similar to issue: [${issue.node.title}](${modifiedUrl}#${issue.node.number})\n\n`);
+  });
+  // Append new footnotes to the body, keeping the previous ones
+  if (footnotes) {
+    updatedBody += "\n\n" + footnotes.join("");
+  }
+  // Update the comment with the modified body
+  await context.octokit.rest.issues.updateComment({
+    owner: payload.repository.owner.login,
+    repo: payload.repository.name,
+    comment_id: commentId,
+    body: updatedBody,
+  });
+}
+
+/**
+ * Checks if a annotate footnote exists in the content.
+ * @param content The content to check for annotate footnotes
+ * @returns True if a annotate footnote exists, false otherwise
+ */
+export function checkIfAnnotateFootNoteExists(content: string): boolean {
+  const footnoteDefRegex = /\[\^(\d+)\^\]: \d+% similar to issue: [^\n]+(\n|$)/g;
```

Comment:
This seems brittle to target strings like this. Any ideas to get more reliable cross plugin output targeting? 

@Shiv810 @gentlementlegen @whilefoo rfc

Reply:
Can you make it?